com.microsoft.mrtk.graphicstools.unity/Editor/Experimental/AmbientOcclusion/AmbientOcclusionTool.cs

```diff
@@ -0,0 +1,324 @@
+
+            _vertexs = mesh.vertices;
+            _normals = mesh.normals;
+            _referenceVertexSamples.Clear();
+            _referenceVertexHits.Clear();
+
+            for (int vi = 0; vi < mesh.vertexCount; vi++)
+            {
+                // Do the work in world space
+                _vertexs[vi] = go.transform.TransformPoint(_vertexs[vi]);
+                _normals[vi] = go.transform.TransformVector(_normals[vi]);
+
+                _hitsInHemisphere.Clear();
+
+                Vector3 averageDir = Vector3.zero;
+
+                var origin = _vertexs[vi] + _normals[vi].normalized * settings._originNormalOffset;
+
+                Physics.queriesHitBackfaces = true;
+
+                for (int ni = 0; ni < settings._samplesPerVertex; ni++)
+                {
+                    var sampleDirection = RandomSampleAboveHemisphere(_normals[vi]);
+
+                    // Visualization: Save samples for gizmo
+                    if (vi == settings._referenceVertexIndex && settings._showSamples)
+                    {
+                        _referenceVertexSamples.Add(sampleDirection);
+                    }
+
+                    if (Physics.Raycast(origin, sampleDirection, out hit, settings._maxSampleDistance))
+                    {
+                        _hitsInHemisphere.Add(hit);
+
+                        // Visualization: Save hits for gizmo
+                        if (vi == settings._referenceVertexIndex)
+                        {
+                            _referenceVertexHits.Add(hit);
+                        }
+                    }
+                    else
+                    {
+                        averageDir += sampleDirection;
+                    }
+                }
+
+                Physics.queriesHitBackfaces = false;
+
+                var avgN = averageDir.normalized;
+                _bentNormalsAo[vi] = new Vector4(avgN.x,
+                                                 avgN.y,
+                                                 avgN.z,
+                                                 1 - ((float)_hitsInHemisphere.Count / settings._samplesPerVertex));
+            }
+
+            mesh.SetUVs(settings._uvChannel, _bentNormalsAo);
+
+            // For the results to be visible to the user
+            // we need to be using the graphics tools standard shader
+            // and the appropriate parameter must be enabled
+
+            if (go.GetComponent<MeshRenderer>() is MeshRenderer meshRenderer)
+            {
+                // No material? No problem.
+                if (meshRenderer.sharedMaterials.Length == 0)
+                {
+                    meshRenderer.sharedMaterial = ConfiguredAoMaterial(meshRenderer.gameObject.name);
+                }
+                foreach (var material in meshRenderer.sharedMaterials)
+                {
+                    // Got non-standard manterials that don't support AO?
+                    if (!IsStandardShader(material))
```

Comment:
You might be able to replace with this: https://github.com/microsoft/MixedReality-GraphicsTools-Unity/blob/main/com.microsoft.mrtk.graphicstools.unity/Runtime/Utilities/StandardShaderUtility.cs#L87

Reply:
Fixed